src/app/Models/Traits/HasRelationshipFields.php

```diff
@@ -71,7 +75,7 @@ public static function isColumnNullable($column_name)
         $table = $instance->getTableWithPrefix();
 
         // MongoDB columns are alway nullable
-        if ($conn->getConfig()['driver'] === 'mongodb') {
+        if (!in_array($conn->getConfig()['driver'], CRUD::getSqlDriverList()) {
```

Comment:
Couldn't we use `driverIsSql` directly?
```suggestion
        if (!CRUD::driverIsSql()) {
```

It's a little bit less efficient since that function will get the connection once again, but does the readability worth it? 😅

Reply:
In this specific case I think efficiency goes on top of readability. 

This function would get called multiple times as the number of columns in database, so it would multiply by x2.

In future we can also improve this by setting the connection/table with prefix etc in CRUD object with one call, that would avoid building the connection so many times during the app life-cycle.

Let me know what you think 👍

Reply:
I agree, those variables are used a lot so it should be initialized with CRUD, we may leave it like that for now.

I was checking some child Traits of CrudTrait, and I noticed there are some performance issues, since I'm not fully aware of all of it we may address it on v4.2 😅

Reply:
Ha! I was going to raise the same issue, but I see you guys already covered this. Ok, agreed 🥳